### DISCUSSION TITLE
Next13 & React Query

### DISCUSSION THREAD

**Discussion by lotusjohn** [vercel/next.js#42279] [Help]
How will Next13 affect the use of RQ? 

Will the data fetching and caching be able to replace RQ or what would RQ provide that would still make it useful with N13?
One guy at Next conf (i forgot his name) said that he believed "things like RQ would still be important" and im looking for more details about this but cant really find any.

**Comment by erikbg7** [comment]
Fetching and caching  changes in Next 13 basically affect Server components, while RQ is thought to be used in Client components.

RQ will work as usual if you use the 'use client' convention to work client side. You should think about the new caching as a replacement for:

- SSG/getStaticProps: `force-cache`, 
- SSR/getServerSideProps: `no-store`
- ISR/revalidate: `next: { revalidate: 100 }`

RQ will be as useful as usual 😸

**Reply by curiouscod3** [reply]
@erikbg7 What about hydration/dehydration? 

For example, this would work with the new fetch API with cache option in the server component?
Because, the RQ example with prefetching/dehydrate is intended to work in getServerSideProps and it is expected to work with server component.

```typescript
export async function getServerSideProps(context) {
  // Check if user is authenticated
  const session = await unstable_getServerSession(context.req, context.res, authOptions)
  // If not, redirect to the homepage
  if (!session) {
    return {
      redirect: {
        destination: "/",
        permanent: false,
      },
    };
  }

  const queryClient = new QueryClient();
  await queryClient.prefetchQuery(['homes', { userId: session?.user?.id }], getMyHomes);

  return {
    props: {
      session,
      dehydratedState: JSON.parse(JSON.stringify(dehydrate(queryClient))),
    }
  }
}
```

**Comment by ryanfangk** [comment]
I tested React-Query in /app/page.js, unfortunately it is stuck in the loading state 
I tested the same code below again in page/index.js and react-query worked as usual

```
const { isLoading, error, data, isFetching } = useQuery(
    'all-users',
    async () => {
      const response = await fetch('http://localhost:8080/users', {
        method: 'GET',
      })
      if (!response.ok) {
        throw new Error('Something went wrong')
      }
      return response.json()
    },
    { refetchOnWindowFocus: false, retry: false }
  )

  if (data) {
    console.log(data)
    return <Box>data found</Box>
  }

  if (isLoading) {
    return (
      <>
        <Heading>Loading</Heading>
        {isFetching ? <div>Refreshing...</div> : null}
      </>
    )
  }

  if (error) {
    return <Heading>An error has occured</Heading>
  }
```

**Comment by RaffoNin** [comment]
I use TRPC with the new app directory. Works well with use client. You can also start a normal TRPC client and use it server side inside of server components.
